File: frontend/src/api/retrieval.ts
import apiClient from './client';

export interface RetrievalResult {
    text: string;
    source: string;
    score: number;
    metadata: Record<string, any>;
}

export interface RetrievalResponse {
    results: RetrievalResult[];
    query_embedding?: number[];
    hyde_document?: string;
}

export const retrievalApi = {
    retrieve: async (botId: string, query: string, topK: number = 5): Promise<RetrievalResponse> => {
        const response = await apiClient.post<RetrievalResponse>(`/api/v1/bots/${botId}/retrieve`, {
            query,
            top_k: topK
        });
        return response.data;
    }
};

File: backend/app/schemas/retrieval.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any

class RetrievalRequest(BaseModel):
    query: str
    top_k: int = 5
    filters: Optional[Dict[str, Any]] = None

class RetrievalResult(BaseModel):
    text: str
    source: str
    score: float
    metadata: Dict[str, Any]
    
class RetrievalResponse(BaseModel):
    results: List[RetrievalResult]
    query_embedding: Optional[List[float]] = None
    hyde_document: Optional[str] = None
